python-lib/ontology_tagger.py

```diff
@@ -108,14 +121,23 @@ def _tokenize_keywords(
             List: The tokenized keywords.
 
         """
-        keywords = [get_keyword(keyword, self.normalize_case) for keyword in keywords]
+        if self.lemmatization:
+            self.tokenizer._activate_components_to_lemmatize(language)
+        keywords = [
+            normalize_case_text(keyword, self.normalize_case) for keyword in keywords
+        ]
         tokenized_keywords = list(
-            self.tokenizer.spacy_nlp_dict[language].tokenizer.pipe(keywords)
+            self.tokenizer.spacy_nlp_dict[language].pipe(keywords)
         )
-        self._keyword_to_tag[language] = {
-            get_keyword(keyword.text, self.normalize_case): tag
-            for keyword, tag in zip(tokenized_keywords, tags)
-        }
+        if self.lemmatization:
+            self._keyword_to_tag[language] = {
+                lemmatize_doc(keyword): tag
+                for keyword, tag in zip(tokenized_keywords, tags)
```

Comment:
Here it seems that the `tokenized_keywords` are in fact `tokenized_docs` (as in spacy's `Doc`). This confused me a bit, especially when applying the `lemmatize_doc` function on a `keyword` which is in fact not a `str` but a `Doc`

Reply:
Thank you, I did not noticed that possible confusion ! I tried to be more clear in this commit 6c3fbf1